auxin/src/lib.rs

```diff
@@ -2024,33 +2046,43 @@ where
 	) -> std::result::Result<(SenderKeyName, SenderKeyDistributionMessage), GroupsError> {
 		let sender_key_name = SenderKeyName {
 			distribution_id: distribution_id.clone(),
-			sender: AuxinDeviceAddress { 
-				address: AuxinAddress::Uuid(self.context.identity.address.get_uuid().cloned().unwrap()),
+			sender: AuxinDeviceAddress {
+				address: AuxinAddress::Uuid(
+					self.context.identity.address.get_uuid().cloned().unwrap(),
+				),
 				device_id: self.context.identity.address.device_id,
 			},
 		};
-		let sender_key_distribution_message = groups::sender_key::create_distribution_message(
+		let sender_key_distribution_message = create_sender_key_distribution_message(
+			&sender_key_name.sender.uuid_protocol_address().unwrap(),
+			*distribution_id,
 			&mut self.context.sender_key_store,
-			&sender_key_name,
 			&mut self.rng,
-			&self.context.ctx,
+			self.context.ctx.get(),
 		)
 		.await?;
+
 		//Make sure we don't lose the information we just set up.
-		let record = process_sender_key(
+		process_sender_key_distribution_message(
+			&sender_key_name.sender.uuid_protocol_address().unwrap(),
+			&sender_key_distribution_message,
```

Comment:
These constitute the major changes. These are two new functions are provided by `libsignal_protocol` that do almost exactly what Auxin was implementing itself. It's now impossible for Auxin to implement these functions by itself because the API it relied on to do so is now private and encapsulated within these two functions.